### DISCUSSION TITLE
How to escape special chars in Assertions?

### DISCUSSION THREAD

**Discussion by ssnorrizaurous** [ovh/venom#384] [General]
I'd like to write assertions such as ShouldContain and ShouldContainSubstring which contain characters such as colons and quotes. For example, below I would like to see that that stderr contains "connect: connection refused".

```- name: bad port
    steps:
      - script: REMOTE_ADDR={{.url}}:9999 ./cli list
        type: exec
        assertions:
          - result.code ShouldEqual 1
          - result.systemerr ShouldContainSubstring "connect: connection refused"```

 Even when quoted Venom returns:

```
error decoding assertions: 1 error(s) decoding:

* 'Assertions[1]' expected type 'string', got unconvertible type 'map[string]interface {}'
```

**Comment by yesnault** [comment]
Hi,
You have to used quote there are some unconvertible type

example:

```yml
name: Title of TestSuite
testcases:
- name: Check if exit code != 1 and echo command response in less than 1s
  steps:
    - script: 'echo "connect: connection refused"'
      assertions:
        - result.code ShouldEqual 0
        - 'result.systemout ShouldContainSubstring "connect: connection refused"'
